.github/workflows/test-report.yml

```diff
@@ -0,0 +1,16 @@
+name: Test Report
+on:
+  workflow_run:
+    workflows: ['CI']                     # runs after CI workflow
+    types:
+      - completed
+jobs:
```

Comment:
Why not just run this as an extra step after exporting test results?

Reply:
Following the instructions on the test-reporter action:
https://github.com/dorny/test-reporter#recommended-setup-for-public-repositories

I haven't tested it for real yet, so not 100% sure it's necessary, but it certainly won't hurt to try as advertised first.

Reply:
I'll keep this comment unresolved and remember to try it out.

Reply:
Re-read this article and confirming the recommended "safe" way to check out PRs from forked repos without allowing the forked repo to have write access to our repo is to run any task that requires write access as a separate workflow where the context will be safe

https://securitylab.github.com/research/github-actions-preventing-pwn-requests/